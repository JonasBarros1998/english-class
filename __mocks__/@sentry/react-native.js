jest.mock('@sentry/react-native', () => ({
  Sentry: {
    captureException: jest.fn(),
    captureMessage: jest.fn()
  }
}));
