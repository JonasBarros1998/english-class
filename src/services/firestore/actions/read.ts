import { List } from "@global/interfaces/Card";
import {connectCollection} from "../connect";
import {databases} from "../constants/collections";

export async function findAll(collections: databases) {
  const lists: List[] = [];

  await connectCollection(collections)
    .get()
      .then(function(response) {
        if (response.empty === false) {
          response.forEach((item) => {
            lists.push(item.data() as List);
          })
        }
        else return lists
        
      })
      .catch(function(error) {
        console.error("ERROR FIND_ALL");
        throw error;
      });
  
  return Promise.resolve(lists);
};

