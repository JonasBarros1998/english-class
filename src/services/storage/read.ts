import AsyncStorage from '@react-native-async-storage/async-storage';

export async function read<Type>(key: string): Promise<Type | null> {
  const item = await AsyncStorage.getItem(key);
  if (item === null) {
    return item;
  }
  return JSON.parse(item);
}