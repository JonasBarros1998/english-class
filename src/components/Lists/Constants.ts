import {Dimensions} from 'react-native';

const WIDTH_SCREEN = Dimensions.get('window').width;

export {WIDTH_SCREEN};
