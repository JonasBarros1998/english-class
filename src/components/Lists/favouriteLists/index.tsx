import React from 'react';

import {FlatList} from 'react-native';

import Card from '../../Cards';

function FavouriteList() {
  /*
  return (
    <FlatList
      renderItem={({item}) => <Card {...item} />}
      keyExtractor={({id}) => String(id)}
    />
  );*/
}

export default FavouriteList;
