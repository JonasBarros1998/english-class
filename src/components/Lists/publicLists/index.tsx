import {userList} from '@global/types/userList';
import React, {useEffect, useCallback, useState} from 'react';
import {FlatList} from 'react-native';
import Card from '../../Cards';
import {loadPublicList} from './useCase/loadPublicList';

function PublicList() {
  const [publicList, setPublicList] = useState<null | userList[]>();

  const loadData = useCallback(() => {
    loadPublicList().then(function (response) {
      setPublicList(response);
    });
  }, []);

  useEffect(() => {
    loadData();
  }, [loadData]);

  return (
    <FlatList
      data={publicList}
      renderItem={({item}) => {
        return <Card {...item} />;
      }}
      keyExtractor={({id}) => String(id)}
    />
  );
}

export default PublicList;
