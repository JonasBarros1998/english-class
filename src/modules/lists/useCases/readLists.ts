import { findAll } from "@services/firestore/actions/read";
import { collections } from "@services/firestore/constants/collections";

export async function findAllLists() {
  return await findAll(collections.lists)
    .then(function(response) {
      return response;
    })
    .catch((error) => {throw error;})
    
}
