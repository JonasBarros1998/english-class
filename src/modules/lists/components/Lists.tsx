import React, { useEffect, useState } from 'react';
import { FlatList, Text, View } from 'react-native';
import { useTheme } from 'react-native-paper';

import { findAllLists } from '../useCases/readLists';
import { List } from '@global/interfaces/Card';

import { styles } from '../styles/cards';

export default function Lists() {
  const [lists, setLists] = useState<List[]>();
  const [refresh, setRefresh] = useState<boolean>(false);

  useEffect(() => {
    findAllLists()
      .then(function(response) {
        setLists([...response]);
      })
  }, []);

  async function refreshList() {
    setRefresh(true);
    findAllLists()
      .then(function(response) {
        setLists([...response]);
      })
      .catch(function() {
        setRefresh(false)
      })
      .finally(() => {
        setRefresh(false);
      });
  }

  const theme = useTheme();
  const css = styles(theme);

  return (
    <View>
      
      <FlatList
        testID='card'
        data={lists}
        refreshing={refresh}
        onRefresh={() => refreshList()}
        renderItem={({item, index}) => {
          return (
            <View style={{...css.container}} testID="content">
              <View style={{...css.card}}>
                <Text style={{
                  ...css.cardInfo, 
                  ...css.subTitle
                  }}>{item.title}</Text>
              </View>
            </View>
          )
        }}
      />
    </View>
  )
}
