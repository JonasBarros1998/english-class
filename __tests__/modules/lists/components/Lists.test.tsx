import React from 'react';
import { render } from '@testing-library/react-native';
import Lists from '@modules/lists/components/Lists';
import { findAllLists } from '@modules/lists/useCases/readLists';
import {findAll} from '../../../../src/services/firestore/actions/read';

jest.mock('../../../../src/services/firestore/actions/read');

jest.mock('../../../../src/modules/lists/useCases/readLists.ts');

jest.mock('@react-native-firebase/firestore', () => ({
  firestore: {
    collection: jest.fn(),
  }
}));

jest.mock('react-native-paper', () => ({
  __esModule: true,
  IconButton: () => 'IconButton',
  useTheme: () => ({
    colors: {
      primary: "black"
    }
  })
}));


const findCards = findAll as any;
findCards.mockImplementation(() => {
  return new Promise((resolve, _) => {
    resolve({
      cardsOfList: [{
        id: '123',
        word: 'card',
        context: 'my card',
        translation: 'cartao'
      }],
      title: 'my first card',
    });
  });
})

const findLists = findAllLists as any;

findLists.mockImplementation(() => {
  return new Promise((resolve, _) => {
    resolve({
      cardsOfList: [{
        id: '123',
        word: 'card',
        context: 'my card',
        translation: 'cartao'
      }],
      title: 'my first card',
    });
  });
});



describe('List all cards', function() {

  test('When load component, should render all cards', function() {

    const component = render(<Lists />).toJSON() as any;
    if (component !== null) {
      expect(component.type).toMatch('View');
    }

  })
});