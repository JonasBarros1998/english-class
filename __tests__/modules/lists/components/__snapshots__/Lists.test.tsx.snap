// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List all cards When load component, should render all cards 1`] = `
<View>
  <RCTScrollView
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="card"
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;
